### DISCUSSION TITLE
How to use Hugging Face Models with Semantic Kernel w/ C#

### DISCUSSION THREAD

**Discussion by mrlife** [microsoft/semantic-kernel#6652] [General]
Hi, are the following sentence transformers available to use with C#?

ref: https://devblogs.microsoft.com/semantic-kernel/how-to-use-hugging-face-models-with-semantic-kernel/

```python
kernel = sk.Kernel()

# Configure LLM service
kernel.config.add_text_completion_service(
    "gpt2", sk_hf.HuggingFaceTextCompletion("gpt2", task="text-generation")
)
kernel.config.add_text_embedding_generation_service(
    "sentence-transformers/all-MiniLM-L6-v2",
    sk_hf.HuggingFaceTextEmbedding("sentence-transformers/all-MiniLM-L6-v2"),
)
kernel.register_memory_store(memory_store=sk.memory.VolatileMemoryStore())
kernel.import_skill(sk.core_skills.TextMemorySkill())
```

I see [HuggingFaceTextEmbeddingGenerationService](https://learn.microsoft.com/en-us/dotnet/api/microsoft.semantickernel.connectors.huggingface.huggingfacetextembeddinggenerationservice?view=semantic-kernel-dotnet), however it doesn't seem like there are equivalent methods in C#, e.g. `Config()`, `ImportSkill()`, etc..

**Comment by rogerbarreto** [comment]
HI @mrlife , thanks for reaching out.

To use Embeddings and Configure the kernel you need to follow the .Net pattern.

Basically, create a builder, configure it and build your kernel.

Follow this example: [https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/Memory/HuggingFace_EmbeddingGeneration.cs](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/Memory/HuggingFace_EmbeddingGeneration.cs)
